src/ClusterProviders/Proto.Cluster.Consul/ConsulProvider.cs

```diff
@@ -0,0 +1,134 @@
+﻿// -----------------------------------------------------------------------
+//   <copyright file="ConsulProvider.cs" company="Asynkron HB">
+//       Copyright (C) 2015-2017 Asynkron HB All rights reserved
+//   </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Consul;
+
+namespace Proto.Cluster.Consul
+{
+    public class ConsulProvider : IClusterProvider
+    {
+        private TimeSpan _blockingWaitTime;
+        private ConsulClient _client;
+        private string _clusterName;
+        private TimeSpan _deregisterCritical;
+        private string _id;
+        private ulong _index;
+        private TimeSpan _refreshTtl;
+        private bool _shutdown;
+        private TimeSpan _ttl;
+
+
+        public async Task RegisterMemberAsync(string clusterName, string host, int port, string[] kinds)
+        {
+            _id = $"{clusterName}@{host}:{port}";
+            _clusterName = clusterName;
+            _index = 0;
+            _ttl = TimeSpan.FromSeconds(3);
+            _refreshTtl = TimeSpan.FromSeconds(1);
+            _deregisterCritical= TimeSpan.FromSeconds(10);
+            _blockingWaitTime = TimeSpan.FromSeconds(20);
+

```

Comment:
Yes, the Consul settings should also be configurable.
I suppose the new .NET core `IOption<T>` is the modern way to go about handling options.